test/test_gpu2.cpp

```diff
@@ -7296,6 +7296,44 @@ TEST_F(NVFuserTest, FusionPredicateElimination8_CUDA) {
       "T6 should not be predicated");
 }
 
+TEST_F(NVFuserTest, FusionPredicateElimination9_CUDA) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  const int M = 1024, split = 512;
+
+  // Algorithm
+  auto tv0 = makeContigTensor(1, DataType::Float);
```

Comment:
I think you have to do
```C++
auto tv0 = makeContigConcreteTensor({M}, DataType::Float);
```
otherwise, it is impossible to prove `T0.size[0]` vs `512` which is larger, so predicate can not be eliminated.

Reply:
You are right, this solved the problem with unexpected predicated in for `tv0c` - now everything works as expected. Thank you!